src/torrent.cpp

```diff
@@ -8547,6 +8527,16 @@ namespace libtorrent
 		stop_announcing();
 	}
 
+	bool torrent::need_save_resume_data() const
+	{
+		// save resume data every 15 minutes regardless, just to
+		// keep stats up to date
+		return m_need_save_resume_data ||
+			// cast ratio<1l, 1000000000l> to seconds_t ratio
```

Comment:
you did ask why it's neccesary to cast a duration between 2 timepoints to seconds. Feel free to suggest another comment.